### PR TITLE
Miscellaneous JS fixes and cleanup

### PR DESCRIPTION
**Pull request by brianhelba**
See individual commit messages for details.

### CODE REVIEW

File: web_external/collection.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            offset -= this.pageLimit;
+            offset = Math.max(0, offset);
+        }
+        this.fetch(_.extend(params, {offset: offset}), false);
+    },
+
+    pageNum: function () {
+        return Math.floor(this._currentOffset() / this.pageLimit);
+    }
+});
```
Comment (msmolens) [inline web_external/collection.js]: My only concern is managing this if the plugin is upgraded for Girder 2.0.

File: web_external/ImagesGallery/Filter/ImagesFacetView.js [hunk 2]
```diff
@@ -246,7 +246,7 @@ isic.views.ImagesFacetHistogramView = isic.views.ImagesFacetView.extend({
         knob.call(d3.behavior.drag()
             .origin(_.bind(function () {
                 return { x: 0, y: knobScale(this.scale.yMax) };
-            }, this)).on('drag', _.bind(function () {
+            }, this)).on('drag', function () {
```
Comment (msmolens) [inline web_external/ImagesGallery/Filter/ImagesFacetView.js]: I don't think these d3 callbacks support passing a context.

File: web_external/ImagesGallery/index.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ isic.views.ImagesView = isic.View.extend({
             parentView: this
         });
 
-        $(window).on('resize.ImagesView', _.bind(this.render, this));
+        $(window).on('resize.ImagesView', this.render, this);
```
Comment (msmolens) [inline web_external/ImagesGallery/index.js]: jQuery event handlers don't support passing a context like this.

File: web_external/common/Viewer/ImageFullscreenWidget.js [hunk 4]
```diff
@@ -2,17 +2,17 @@ isic.views.ImageFullscreenWidget = isic.View.extend({
...
             model: this.model
-        })).girderModal(this).on('shown.bs.modal', _.bind(function () {
+        })).girderModal(this).on('shown.bs.modal', function () {
             this.imageViewerWidget = new isic.views.ImageViewerWidget({
                 el: this.$('.isic-image-fullscreen-container'),
                 model: this.model,
                 parentView: this
             }).render();
-        }, this)).on('hidden.bs.modal', _.bind(function () {
+        }, this).on('hidden.bs.modal', function () {
```
Comment (msmolens) [inline web_external/common/Viewer/ImageFullscreenWidget.js]: jQuery event handlers don't support passing a context like this.

File: web_external/ImagesGallery/Filter/ImagesFacetView.js (hunk 2 above)
Comment (brianhelba) [inline web_external/ImagesGallery/Filter/ImagesFacetView.js]: You're right, this was a find-replace oversight.

File: web_external/ImagesGallery/index.js (hunk 3 above)
Comment (brianhelba) [inline web_external/ImagesGallery/index.js]: Fixing.

File: web_external/common/Viewer/ImageFullscreenWidget.js (hunk 4 above)
Comment (brianhelba) [inline web_external/common/Viewer/ImageFullscreenWidget.js]: Fixing.

File: web_external/collection.js (hunk 1 above)
Comment (brianhelba) [inline web_external/collection.js]: True, we may have to rewrite some of this for a migration, but having to individually hack every call to `fetch` is becoming unmaintainable, particularly when we want to use pagination with other potentially-changing parameters.

Eventually, I'd like to upstream some of these fixes, but that will have to be after the 2.0 migration.

**Pull request merged**